### PR TITLE
Feature/8347 react intl prepping for upgrade

### PR DESCRIPTION
**Pull request by adamhooper**
Wire up material-ui RTL. Nix isRtl. [#8347] [#8249]

Changes an i18n message! The `<Footer>` message has changed so it doesn't end with `"by"`: it ends with `"by {meedan}"`. (This avoids string concatenation -- a no-no in i18n.)

Lessons of the day:

* We overwrote a couple of material-ui defaults here. For instance, `<Drawer>` will appear on the correct side by default, and `rtlDetect` was giving us the wrong logic. (Also, we weren't setting the drawer width correctly -- fixed!)
* `text-align: start` and `text-align: end` align buttons perfectly
* `flex-grow: 1` aligns block contents perfectly

`<Home>`: handle `<FormattedMessage>` in flash messages.

`<TeamHeader>`: nix `onClick`, which modified `<Home>` `state.open`. Rewrite without styled-components. (I had an extremely hard time actually _rendering_ `<TeamHeader>`. I removed a redirect from `CheckNetworkLayer.js` and browsed to a media with a valid ID (because an _invalid_ ID would lead to redirect).)

`<TeamInviteCard>`: don't use RTL [#8347]. Also, use material-ui CSS [#8249]

`<TeamMembers>`: revamp. Eradiate `<Offset>` and `isRtl` [#8347]

Update `react-icons` (for Slack) and use new `<SocialIcon>` throughout. Fixes a failing prop-type. [#8179]

`<Offset>` ... I don't think anybody understood what it did. It was often added as the last item of an align-left container, and all it did was add a padding-right. It depended on `isRtl`, so it had to go. [#8347] [#8249]

To nix `<Offset>` from `<TeamMembers>`, I used material-ui's list-item components [#8249] ... and I noticed that the `<RCTooltip>` was putting tooltips in the wrong place because (A) they were exactly where the mouse goes when switching between users; and (B) they would _load_ and then switch size after loading. I re-examined the Relay code there. This should make it easy to switch to Relay Modern [#6211].

`<AutoCompleteMediaItem>`: nix `intl` and fix races.

Previously, two autocomplete HTTP requests could race one another. It looks like the component used a 2-second search timeout to avoid most races -- hurting usability and not even fixing the problem.

Now, changing search text aborts the previous autocomplete.

Also, the "Loading" and "Error" messages are built-in to the component now: that's part of the material-ui `<AutoComplete>` API.

`<MoveDialog>`: intl [#8347], Relay [#6221].

* Disable unused, untested API for choosing from a different team.
* `<DestinationProjects>`: use `<Autocomplete>` instead of react-select. This fixes a warning [#8179] and cleans styles [#8249].
* Make everything a fragment container [#6211]. Reduces code spaghetti; nixes a dependence on `CheckContext`; saves an HTTP request.

### CODE REVIEW

File: src/app/components/team/TeamInviteCard.js [hunk 1]
```diff
@@ -1,27 +1,66 @@
...
+//   background-image: url("${encodeSvgDataUri(teamInviteSvg)}");
+//   background-repeat: no-repeat;
+//   background-size: contain;
+//   background-position: top ${props => props.theme.dir === 'rtl' ? 'right' : 'left'};
+//   padding-${props => props.theme.dir === 'rtl' ? 'right' : 'left'}: ${units(21)};
+//   @media all and (max-width: ${breakpointMobile}) {
+//     padding-${props => props.theme.dir === 'rtl' ? 'right' : 'left'}: ${units(12)};
+//     background-size: ${units(10)};
+//   }
+// `;
```
Comment (caiosba) [inline src/app/components/team/TeamInviteCard.js:63]: Can we remove this block?

File: src/app/relay/mutations/UpdateTeamUserMutation.js [hunk 2]
```diff
@@ -7,59 +7,67 @@ class UpdateTeamUserMutation extends Relay.Mutation {
...
+        }
+      }
+    `,
+  };
+
   getFatQuery() {
     return Relay.QL`
       fragment on UpdateTeamUserPayload {
-        team,
+        team_userEdge
```
Comment (caiosba) [inline src/app/relay/mutations/UpdateTeamUserMutation.js:37]: I don't see it being used on `getConfigs`... do we need it?

File: src/app/components/media/MediaActionsBar.js [hunk 3]
```diff
@@ -262,6 +242,8 @@ class MediaActionsBarComponent extends Component {
           media,
           metadata: JSON.stringify(embed),
           id: media.id,
+          srcProj: null,
+          dstProj: null,
```
[the remark this answers is not in the document]
Comment (adamhooper) [reply src/app/components/media/MediaActionsBar.js:246]: They're in the mutation's fragments. When we omit them, Relay Modern warns.

The better solution is to code a separate mutation per operation. I want to work on that in the near future.

(Perhaps `check-api` ought to implement a separate mutation per operation, too. The existing input/payload schemas make it impossible to optimistically update -- and indeed impossible for the server to update project-list counts. For example, moving from multiple lists [from the 'All Items' search results] to a separate list, our existing signatures don't let us update the project counts for the moved-from projects.)

[the remark this answers is not in the document]
Comment (amoedoamorim) [reply src/app/components/media/MediaActionsBar.js:246]: Thanks for clarifying.

> The better solution is to code a separate mutation per operation. I want to work on that in the near future.

^ I like this

**Pull request merged**